app/src/main/java/com/ss20/se2/monopoly/view/GameboardActivity.java

```diff
@@ -137,42 +152,113 @@ protected void onCreate(Bundle savedInstanceState) {
 		view_position = findViewById(R.id.number_playerposition);
 		view_balance = findViewById(R.id.text_balance);
 		updateBalance = findViewById(R.id.changeOfBalance);
+		altbutton = findViewById(R.id.altbutton);
 
 
 		button_rollDice.setOnClickListener(new View.OnClickListener() {
 			@Override
 			public void onClick(View v) {
 
-				roll1 = dice.roll();
-				roll2 = dice2.roll();
-				amount = roll1 + roll2;
-				checkDouble(roll1, roll2);
+				Toast.makeText(getApplicationContext(), "Shake or tap to roll!", Toast.LENGTH_SHORT).show();
 
-				if (checkDouble(roll1, roll2)){
-					doublescounter++;
-				}else{
-					doublescounter = 0;
-				}
-
-				if(doublescounter == 3){
-					//Move to jail
-				}
+				mSensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+				accel = 9f;
+				currentaccel = SensorManager.GRAVITY_EARTH;
+				lastaccel = SensorManager.GRAVITY_EARTH;
+				altbutton.setVisibility(View.VISIBLE);
 
-				setOldBalance(currentPlayer.getBalance());
+				final SensorEventListener mSensorListener = new SensorEventListener(){
 
-				gameboard.move("Player 1", amount);
+					@Override
+					public void onSensorChanged(SensorEvent event){
+
+						float x = event.values[0];
+						float y = event.values[1];
+						float z = event.values[2];
+
+						lastaccel = currentaccel;
+						currentaccel = (float) Math.sqrt((double) (x * x + y * y + z * z));
+						float delta = currentaccel - lastaccel;
+						accel = accel * 0.9f + delta;
+
+						if(accel > 11){
+
+							mSensorManager.unregisterListener(this);
+							altbutton.setVisibility(View.GONE);
+
+							roll1 = dice.roll();
+							roll2 = dice2.roll();
+							amount = roll1 + roll2;
+
+							checkDouble(roll1, roll2);
+							if (checkDouble(roll1, roll2)){
+								doublescounter++;
+							}else{
+								doublescounter = 0;
+							}
+							if (doublescounter == 3){
+								//Move to jail
+							}
+							setOldBalance(currentPlayer.getBalance());
+							gameboard.move("Player 1", amount);
+
+							updateBalance.setText("");
+							currentPlayer.setCurrentPosition(gameboard.getPosition("Player 1"));
+							checkPlayersPosition(currentPlayer);
+
+							view_numberDice.setText("Roll 1: " + Integer.toString(roll1));
+							view_numberDice2.setText("Roll 2: " + Integer.toString(roll2));
+							view_position.setText(Integer.toString(gameboard.getPosition("Player 1")));
+							findViewById(R.id.playericon).setX(fields[gameboard.getPosition("Player 1")].getX());
+							findViewById(R.id.playericon).setY(fields[gameboard.getPosition("Player 1")].getY());
```

Comment:
This should be extracted in to a method (e.g. processRoll) to prevent problems with merging.